### PR TITLE
Improvements to code generators

### LINKED ISSUE #2847: Generator improvements
Currently, all generators require a language flag of `--python` or `--ruby` to be provided, even when it does not make sense. This does not match the documentation page for [Code Generators](https://docs.openc3.com/docs/getting-started/generators). The language flag should only be required for generators that actually use it. 

In some cases, providing the language flag _without_ a NAME parameter allows the code generator to work, using the language as the name (ie: plugin, tool)

- Consider accepting a language flag for plugin generation that would then be used as the default language for future code generation within said plugin if language flag is not provided (just as a default, could be overridden)
- Consider for any generator that requires providing a target, using said target's language as the default (could be overridden) if language flag is not provided

Update related documentation as appropriate, add tests for edge cases of code generators

### PR DESCRIPTION
**Pull request by EmilyRagan**
Closes #2847

### CODE REVIEW

File: openc3/lib/openc3/utilities/cli_generator.rb [hunk 1]
```diff
@@ -115,30 +356,35 @@ def self.process_template(template_dir, the_binding)
...
+            '',
+            'cli generate plugin demo',
+            'Creates: openc3-cosmos-demo/ with no language default',
+          ],
+          is_plugin: true,
+          docs: 'https://docs.openc3.com/docs/configuration/plugins',
+          exit_code: (args[1].nil? ? 1 : 0),
+        )
+      end
+      if args.length < 2 or args.length > 2
```
Comment (clayandgen) [inline openc3/lib/openc3/utilities/cli_generator.rb]: is this just `args.length != 2` ?

File: openc3/lib/openc3/utilities/cli_generator.rb [hunk 2]
```diff
@@ -65,34 +77,263 @@ def self.generate(args)
...
+
+      if lang_flags.size > 1 && lang_flags.uniq.size > 1
+        abort("Cannot specify both --ruby and --python.")
+      end
+
+      # If a language flag is present but no positional NAME follows the generator name,
+      # the user wrote something like `cli generate plugin --python`. Abort with guidance.
+      unless flag_indices.empty?
+        first_flag_idx = flag_indices.min
+        has_positional_name = args[1...first_flag_idx].any? { |a| a != '--ruby' && a != '--python' }
```
Comment (clayandgen) [inline openc3/lib/openc3/utilities/cli_generator.rb]: Isn't `first_flag_idx` always 1 if it makes it to this part of the code? 

Not sure if this level of guidance is strictly necessary, if the `--help` can show uses

File: openc3/lib/openc3/utilities/cli_generator.rb [hunk 3]
```diff
@@ -521,223 +709,60 @@ def self.generate_tool(args)
...
+        template: 'limits_response',
+        template_source: 'response', # template file is response.rb / response.py
+        class_var: 'response_class',
+        usage_intro: 'To use the limits response add the following to a telemetry item:',
+        usage_directive: 'LIMITS_RESPONSE %{basename}',
+        docs: 'https://docs.openc3.com/docs/configuration/telemetry#limits_response',
+        help_example: 'CUSTOM',
+      })
     end
-
```
Comment (clayandgen) [inline openc3/lib/openc3/utilities/cli_generator.rb:687]: Consider adding back this empty line for readability? Same for `generate_limits_response`, `generate_processor`, `generate_conversion`

File: openc3/lib/openc3/utilities/cli_generator.rb (hunk 1 above)
Comment (EmilyRagan) [reply openc3/lib/openc3/utilities/cli_generator.rb]: lol good catch

File: openc3/lib/openc3/utilities/cli_generator.rb (hunk 2 above)
Comment (EmilyRagan) [reply openc3/lib/openc3/utilities/cli_generator.rb]: Good point, it is always 1. Simplifying.

I've run into edge cases in the current implementation where if the language option is provided without name (in effect, before name), especially for things that don't actually _need_ a language, it'll use the language as the name, so I wanted to make sure that the edge case is prevented

**Review by clayandgen — APPROVED** [review]
